lib/src/data.dart

```diff
@@ -0,0 +1,14 @@
+// part of '../animated_icon_button.dart';
+import 'package:flutter/material.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'data.freezed.dart';
+
+@freezed
+abstract class AnimatedIconButtonEntry with _$AnimatedIconButtonEntry {
+  const factory AnimatedIconButtonEntry(
+      {required Icon icon,
+      @Default(Colors.transparent) Color backgroundColor,
+      String? tooltip,
```

Comment:
For what used this field ?

Reply:
you mean the whole class?

Reply:
No. I write about 
```dart
String? tooltip
```

Reply:
it depends how you want to implement tooltip.
If you want one tooltip for the whole widget = no. but how I implemented it, the tooltip changes with the prominent icon, then it is needed here

Reply:
Sorry, I didn't notice that this field is in used.

Reply:
ikr, it is a little hidden at the end of the file:
`tooltip: nowIcon.tooltip,`